### PR TITLE
Refactor octree generation

### PR DESCRIPTION
**Pull request by ksavinash9**
Refactor octree generation. This will helps in:
* Making it extensible for building octree from file or iterator
* Flexibility to use octree builder in other packages

Future plans:
* Current octree generation algorithm takes at-most 2.8 times the size of point clouds on disk. Hence, there's a need to optimize it by using reservoir sampling algorithm.

### CODE REVIEW

File: src/bin/build_octree.rs [hunk 1]
```diff
@@ -12,252 +12,12 @@
...
-use point_viewer::octree;
-use point_viewer::ply::PlyIterator;
-use point_viewer::proto;
-use point_viewer::pts::PtsIterator;
-use protobuf::Message;
-use scoped_pool::{Pool, Scope};
-use std::fs::{self, File};
-use std::io::{BufWriter, Stdout};
 use std::path::PathBuf;
-use std::sync::mpsc;
```
Comment (SirVer) [inline src/bin/build_octree.rs]: remove empty line

**Comment by SirVer** [comment]
Please change the PR description: what did we refactor? why do we do this, i.e. the plan for going forward.

**Comment by ksavinash9** [comment]
@SirVer Please, merge.

**Comment by SirVer** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user SirVer. Merge queue now has a length of 1.

**Pull request merged**